src/app/bav/controllers/root.js

```diff
@@ -1,14 +1,40 @@
 const { Controller: BaseController } = require("hmpo-form-wizard");
+const { API } = require("../../../lib/config");
+const rsa = require("node-rsa");
 
 class RootController extends BaseController {
   async saveValues(req, res, next) {
-    //TODO: Call new fetchNameInfo Lambda to users name as single string
-    // req.sessionModel.set("fullName", data.full_name);
+    try {
+      const encryptedJSON = await this.getNameInfo(req.axios, req);
+      const key = await this.getDecryptKey(req.axios, req);
+      const decryptKey = new rsa(key);
+      const decryptedJSON = decryptKey.decrypt(encryptedJSON, "utf8");
+      const decryptedObject = JSON.parse(decryptedJSON);
+      const decryptedName = decryptedObject.name;
 
-    req.sessionModel.set("fullName", "John Peter Smith");
+      req.sessionModel.set("fullName", decryptedName);
+    } catch (error) {
+      console.log(error);
```

Comment:
Do you think we could add some context to this? Like console.log("Failed to fetch or decrypt full name", error)

Reply:
Done